src/System.Windows.Forms/src/System/Windows/Forms/FolderBrowserDialog.cs

```diff
@@ -205,12 +205,25 @@ protected override bool RunDialog(IntPtr hWndOwner)
 
         private bool TryRunDialogVista(IntPtr owner, out bool returnValue)
         {
-            OpenFileDialog.NativeFileOpenDialog dialog;
+            IFileOpenDialog dialog;
             try
             {
                 // Creating the Vista dialog can fail on Windows Server Core, even if the
                 // Server Core App Compatibility FOD is installed.
-                dialog = new OpenFileDialog.NativeFileOpenDialog();
+                Guid IID_IFileOpenDialog = new Guid("d57c7288-d4ad-4768-be02-9d969532d960");
+                Guid IID_FileOpenDialogCoClass = new Guid("DC1C5A9C-E88A-4dde-A5A1-60F82A20AEF7");
```

Comment:
This is actually a class not an interface

```suggestion
                Guid CLSID_FileOpenDialog = new Guid("DC1C5A9C-E88A-4dde-A5A1-60F82A20AEF7");
```

Reply:
Also apply this change in the other places.